File: SDKs/amazonaws.com/rekognition/2016-06-27/java/lib/src/main/java/openapisdk/models/shared/StartTechnicalCueDetectionFilter.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * StartTechnicalCueDetectionFilter
 * Filters for the technical segments returned by <a>GetSegmentDetection</a>. For more information, see <a>StartSegmentDetectionFilters</a>.
**/
public class StartTechnicalCueDetectionFilter {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("BlackFrame")
    public BlackFrame blackFrame;
    public StartTechnicalCueDetectionFilter withBlackFrame(BlackFrame blackFrame) {
        this.blackFrame = blackFrame;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("MinSegmentConfidence")
    public Float minSegmentConfidence;
    public StartTechnicalCueDetectionFilter withMinSegmentConfidence(Float minSegmentConfidence) {
        this.minSegmentConfidence = minSegmentConfidence;
        return this;
    }
}

File: SDKs/amazonaws.com/rekognition/2016-06-27/python/src/sdk/models/shared/starttechnicalcuedetectionfilter.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import blackframe as shared_blackframe


@dataclass_json
@dataclasses.dataclass
class StartTechnicalCueDetectionFilter:
    r"""StartTechnicalCueDetectionFilter
    Filters for the technical segments returned by <a>GetSegmentDetection</a>. For more information, see <a>StartSegmentDetectionFilters</a>.
    """
    
    black_frame: Optional[shared_blackframe.BlackFrame] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('BlackFrame') }})
    min_segment_confidence: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('MinSegmentConfidence') }})
    
